transitions/index.js

```diff
@@ -419,11 +419,43 @@ const getMetaData = callback =>
       if (err.statusCode !== 404) {
         return callback(err);
       }
-      doc = {
-        _id: METADATA_DOCUMENT,
-        processed_seq: 0
-      };
+
+      // Doc doesn't exist.
+      // Maybe we have the doc in the old location?
+      return db.medic.get('sentinel-meta-data', (err, doc) => {
+        if (err) {
+          if (err.statusCode !== 404) {
+            return callback(err);
+          }
+
+          // No doc at all, create and return default
+          doc = {
+            _id: METADATA_DOCUMENT,
+            processed_seq: 0
+          };
+          return callback(null, doc);
+        }
+
+        // Old doc exists, delete it and return the base doc to be saved later
+        const stub = {
+          _id: doc._id,
+          _rev: doc._rev,
+          _deleted: true
+        };
+        logger.info('Deleting old metadata document', doc);
+        db.medic.insert(stub, err => {
+          if (err) {
+            return callback(err);
+          }
+
+          doc._id = METADATA_DOCUMENT;
+          delete doc._rev;
+          callback(null, doc);
+        });
+      });
     }
+
+    // Doc exists in correct location
     callback(null, doc);
```

Comment:
Could this be moved to the top of the function?  E.g.
```
if(!err) {
  // Doc exists in correct location
  return callback(null, doc);
}
```
I think this would make it much more obvious what's going on.

You could probably remove two levels of indents this way:
```
if(!err) {
  // Doc exists in correct location
  return callback(null, doc);
}
if(err.status !== 404) {
  return callback(err);
}
```
…handle 404…